src/posix/platform/trel_udp6.cpp

```diff
@@ -405,6 +413,102 @@ static void ReceivePacket(int aSocket, otInstance *aInstance)
     }
 }
 
+static void processNetifAddrEvent(struct nlmsghdr *aNetlinkMessage)
+{
+    struct ifaddrmsg *ifaddr = reinterpret_cast<struct ifaddrmsg *>(NLMSG_DATA(aNetlinkMessage));
+    size_t            rtaLength;
+    otError           error = OT_ERROR_NONE;
+
+    VerifyOrExit(ifaddr->ifa_index == static_cast<unsigned int>(sInterfaceIndex) && ifaddr->ifa_family == AF_INET6);
+
+    rtaLength = IFA_PAYLOAD(aNetlinkMessage);
+
+    for (struct rtattr *rta = reinterpret_cast<struct rtattr *>(IFA_RTA(ifaddr)); RTA_OK(rta, rtaLength);
+         rta                = RTA_NEXT(rta, rtaLength))
+    {
+        if (rta->rta_type != IFA_ADDRESS)
+        {
+            continue;
+        }
+
+        {
+            otIp6Address addr;
+            memcpy(&addr, RTA_DATA(rta), sizeof(addr));
+            bool isInterfaceAddress = memcmp(&sInterfaceAddress, RTA_DATA(rta), sizeof(sInterfaceAddress)) == 0;
```

Comment:
Move bool variable declaration before `memcpy()` and next to `addr`? (so all variables are at the top of scope)